pkg/cloud/services/compute/instance.go

```diff
@@ -44,6 +44,39 @@ const (
 	timeoutInstanceDelete       = 5 * time.Minute
 )
 
+type OpenStackError struct {
```

Comment:
if this is general, I think we should put into util or somewhere else?

Reply:
Yes. I only threw it in here for convenience. @seanschneeweiss would something like this address the fatal vs retriable errors issue (if used consistently).

Reply:
This would most certainly be a solution to the problem, but as indicated in my [comment](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1116#issuecomment-1022047685) I hoped for an easier solution where we just reduce the error cases that call `handleUpdateMachineError`.